flowzone.yml

```diff
@@ -1271,78 +1278,96 @@ jobs:
+                  encoding: 'base64'
+                });
+
+                entries.push({
+                  path: file,
+                  mode: '100644',
+                  type: 'blob',
+                  sha: blob.sha
+                });
+              }
+              return entries;
+            };
+
+            // Get base tree SHA
+            const parentSha = process.env.PARENT_COMMIT_SHA;
+            const baseTreeSha = execSync(`git show -s --format=%T "${parentSha}"`).toString().trim();
+
+            // Create new tree
+            const treeEntries = await createTreeEntries();
+            const { data: tree } = await github.rest.git.createTree({
+              ...context.repo,
+              tree: treeEntries,
+              base_tree: baseTreeSha
+            });
 
-          echo "$response" | jq .
-          echo "sha=$(echo "$response" | jq -r .sha)" >> "${GITHUB_OUTPUT}"
-          echo "json=$(echo "$response" | jq -c .)" >> "${GITHUB_OUTPUT}"
+            return tree;
 
-      # https://docs.github.com/en/rest/git/commits?apiVersion=2022-11-28#create-a-commit
+      # https://octokit.github.io/rest.js/v21/#git-create-commit
+      # https://docs.github.com/en/rest/git/commits#create-a-commit
       - name: Create commit object
         if: inputs.disable_versioning != true
         id: create_commit
         env:
-          GH_TOKEN: ${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}
           MESSAGE: ${{ steps.versionist.outputs.tag }}
+          TREE_SHA: ${{ fromJSON(steps.create_tree.outputs.result).sha }}
           PARENT_COMMIT_SHA: ${{ steps.git_describe.outputs.sha }}
-        run: |
-          response="$(gh api -X POST "/repos/$GH_REPO/git/commits" \
-            -F "message=$MESSAGE" \
-            -F "tree=${{ steps.create_tree.outputs.sha }}" \
-            -F "parents[]=$PARENT_COMMIT_SHA")"
-
-          echo "$response" | jq .
-          {
-            echo "sha=$(echo "$response" | jq -r .sha)" >> "${GITHUB_OUTPUT}" ;
-            echo "json=$(echo "$response" | jq -c .)" >> "${GITHUB_OUTPUT}" ;
-            echo "bot=$(echo "$response" | jq -r .author.name | sed 's/\[bot\]//g')" ;
-          } >> "${GITHUB_OUTPUT}"
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7
+        with:
+          github-token: ${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}
+          result-encoding: json
+          script: |
+            const { data: commit } = await github.rest.git.createCommit({
+              ...context.repo,
+              message: process.env.MESSAGE,
+              tree: process.env.TREE_SHA,
+              parents: [process.env.PARENT_COMMIT_SHA]
+            });
+            core.setOutput('author', commit.author.name);
+            core.setOutput('sha', commit.sha);
+            return commit;
 
       # create a new tag object
       - <<: *createTagObject
+        if: inputs.disable_versioning != true
```

Comment:
Why wasn't this needed before? Was it a bug?

Reply:
It used to be on line 72, as part of the anchor. It makes more sense to have the condition where it is used.